server/auth_server.go

```diff
@@ -503,11 +504,25 @@ func UpdateConfig(authConfig model.AuthConfig) error {
 	commonSettings[providerNameSetting] = authConfig.Provider
 	commonSettings[providerSetting] = authConfig.Provider
 	commonSettings[externalProviderSetting] = "true"
+	err = updateCommonSettings(commonSettings)
```

Comment:
You used to do just a single call to updateCommonSettings. Why call it three separate times now?